surfer/viz.py

```diff
@@ -2178,19 +2179,16 @@ def save_movie(self, fname, time_dilation=4., tmin=None, tmax=None,
             raise ValueError("tmin=%r is smaller than the first time point "
                              "(%r)" % (tmin, self._times[0]))
 
+        # find indexes at which to create frames
+        tstep = 1. / (framerate * time_dilation)
         if tmax is None:
             tmax = self._times[-1]
-        elif tmax >= self._times[-1]:
+        elif tmax > self._times[-1]:
             raise ValueError("tmax=%r is greater than the latest time point "
                              "(%r)" % (tmax, self._times[-1]))
-
-        # find indexes at which to create frames
-        tstep = 1. / (framerate * time_dilation)
-        if np.allclose((tmax - tmin) % tstep, 0):
-            tstop = tmax + tstep / 2.
-        else:
-            tstop = tmax
-        times = np.arange(tmin, tstop, tstep)
+        times = np.arange(tmin, tmax + sys.float_info.epsilon, tstep)
```

Comment:
I don't think this guarantees what you think it does... epsilon should be the delta relative to 1, so if times steps are e.g. in the thousands this will not work. A cleaner way to do it is probably to do `np.arange(n_pts) * tstep`, then you're guaranteed the proper length.